slam-libraries/viam-cartographer/src/slam_service/slam_service.h

```diff
@@ -72,6 +72,9 @@ class SLAMServiceImpl final : public SLAMService::Service {
     // MapBuilder parameters.
     void OverwriteMapBuilderParameters();
 
+    // PaintMap paints the map in jpeg format
+    void PaintMap(std::string &jpeg_img);
```

Comment:
[nit] It's more idiomatic to return a string than to use an out-parameter.

Reply:
Initially, I returned a string but then thought that using the out-parameter would improve performance (which is really bad in the case of`PaintMap`) by removing one more copying step of the string. Am I wrong about this assumption? How can I implement `PaintMap` here by returning a string in a way that optimizes performance?

Reply:
If `PaintMap` constructs a string and then returns it at the end of the function, the string will not be copied when it is returned, due to return value optimization (https://en.cppreference.com/w/cpp/language/copy_elision).